plugin/forward/connect.go

```diff
@@ -18,9 +18,17 @@ import (
 
 func (p *Proxy) readTimeout() time.Duration {
 	rtt := time.Duration(atomic.LoadInt64(&p.avgRtt))
+
 	if rtt < timeout/2 {
 		return 2 * rtt
 	}
+	if rtt < minTimeout {
```

Comment:
the previous condition is less strict, so this condition will never trigger. Let's make it consistent, i.e
```
	if rtt < minTimeout {
		return minTimeout
	}
	if rtt < maxTimeout/2 {
		return 2 * rtt
	}
	return maxTimeout

```